tensorflow/compiler/xla/service/gpu/ir_emission_utils.cc

```diff
@@ -234,6 +234,37 @@ bool IsReductionFromOrToContiguousDimensions(const HloInstruction& reduce) {
   return reduction_dimensions.dimensions[1] >= kWarpSize;
 }
 
+bool IsInputFusibleSlices(const HloInstruction& unnested_hlo,
+                          bool verify_no_strides) {
+  if (!unnested_hlo.IsInputFusion()) {
+    return false;
+  }
+
+  auto is_non_strided = [](const std::vector<int64>& strides) -> bool {
+    return absl::c_all_of(strides, [](int stride) { return stride == 1; });
+  };
+
+  const auto root = unnested_hlo.fused_expression_root();
```

Comment:
Could we spell out the type explicitly here?

Reply:
Done.